src/spaces/compartment.jl

```diff
@@ -0,0 +1,188 @@
+  {A<:AbstractAgent}
+  prev = model.space.grid.s[pos2cell(a)...]
+  ai = findfirst(i -> i == a.id, prev)
+  deleteat!(prev, ai)
+  return a
+end
+
+function move_agent!(a::A, pos::Tuple, model::ABM{A,<:CompartmentSpace}) where 
+  {A<:AbstractAgent}
+  remove_agent_from_space!(a, model)
+  a.pos = pos
+  add_agent_to_space!(a, model)
+end
+
+"""
+    move_agent!(agent::A, model::ABM{A, CompartmentSpace}, dt = 1.0)
+Propagate the agent forwards one step according to its velocity,
+_after_ updating the agent's velocity (see [`CompartmentSpace`](@ref)).
+Also take care of periodic boundary conditions.
+
+For this continuous space version of `move_agent!`, the "evolution algorithm"
+is a trivial Euler scheme with `dt` the step size, i.e. the agent position is updated
+as `agent.pos += agent.vel * dt`.
+
+Notice that if you want the agent to instantly move to a specified position, do
+`move_agent!(agent, pos, model)`.
+"""
+function move_agent!(agent::A, model::ABM{A, <: CompartmentSpace}, dt::Real = 1.0) where {A <: AbstractAgent}
+  model.space.update_vel!(agent, model)
+  pos = agent.pos .+ dt .* agent.vel
+  if model.space.periodic
+    pos = mod.(pos, model.space.dims)
+  end
+  move_agent!(agent, pos, model)
+  return agent.pos
+end
+
+"""
+    move_agent!(agent::A, model::ABM{A, CompartmentSpace}, vel::NTuple{D, N}, dt = 1.0)
+Propagate the agent forwards one step according to `vel` and the model's space, with `dt` as the time step. (`update_vel!` is not used)
+"""
+function move_agent!(agent::A, model::ABM{A,S,F,P}, vel::NTuple{D, X}, dt = 1.0) where {A <: AbstractAgent, S <: CompartmentSpace, F, P, D, X <: AbstractFloat}
+  pos = agent.pos .+ dt .* vel
+  if model.space.periodic
+    pos = mod.(pos, model.space.dims)
+  end
+  move_agent!(agent, pos, model)
+  return agent.pos
+end
+
+#######################################################################################
+# %% Neighbors and stuff
+#######################################################################################
+
+grid_space_neighborhood(α, model::ABM{<:AbstractAgent, <:CompartmentSpace}, r) =
+  grid_space_neighborhood(α, model.space.grid, r)
+
+function nearby_ids(pos::ValidPos, model::ABM{<:AbstractAgent,<:CompartmentSpace}, r = 1)
+  nn = grid_space_neighborhood(CartesianIndex(pos), model, r)
+  s = model.space.grid.s
+  Iterators.flatten((s[i...] for i in nn))
+end
+
+function nearby_positions(pos::ValidPos, model::ABM{<:AbstractAgent,<:CompartmentSpace}, r = 1)
+  nn = grid_space_neighborhood(CartesianIndex(pos), model, r)
+  Iterators.filter(!isequal(pos), nn)
+end
+
+function positions(model::ABM{<:AbstractAgent,<:CompartmentSpace})
+  x = CartesianIndices(model.space.grid.s)
+  return (Tuple(y) for y in x)
+end
+
+function ids_in_position(pos::ValidPos, model::ABM{<:AbstractAgent,<:CompartmentSpace})
+  return model.space.grid.s[pos...]
+end
+
+cell_center(pos::Tuple) = getindex.(modf.(pos), 2) .+ 0.5
```

Comment:
I think this will lead to problems for continuous spaces that cover negative numbers. Depending on the application, it might be sensible for a user that their continuous space spans -10 to 10. We know how to handle these cases because of work we have done in DynamicalSystems.jl, and I will suggest a solution tonight.

Reply:
Yes, I didn't consider negative spaces.

Reply:
Previously, we only allowed positive spaces. I am not sure if negative spaces are needed.